### PR TITLE
UCP/REQUEST: Remove 'const' qualifier from send.buffer

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: src/ucp/rma/rma_send.c [hunk 1]
```diff
@@ -102,7 +102,7 @@ ucp_rma_request_init(ucp_request_t *req, ucp_ep_h ep, const void *buffer,
 {
     req->flags                = flags; /* Implicit release */
     req->send.ep              = ep;
-    req->send.buffer          = buffer;
+    req->send.buffer          = (void*)buffer;
```
[the remark this answers is not in the document]
Comment (yosefe) [reply src/ucp/rma/rma_send.c]: i prefer to keep it in API, so the API would be "const-correct"
internally, we could use a union, but it's not worth the mess IMHO

**Comment by yosefe** [comment]
@shamisp it belongs to the "send" portion of the request because it's filled during "send" operations, are you suggesting to change request layout?

**Comment by shamisp** [comment]
@yosefe I was thinking just about the naming. The "send" buffer for "get" operation is not the "send" but "receive" buffer, and this is why you remove the const ?

**Comment by yosefe** [comment]
yes, this is the reason

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7532/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
Probably we just should document it somewhere that send request is also get
request.

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7535/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7540/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7545/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7546/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shamisp fixed

**Pull request merged**